modular_skyrat/modules/modular_implants/code/nifs_tgui.dm

```diff
@@ -0,0 +1,113 @@
+/obj/item/organ/internal/cyberimp/brain/nif
+	///Currently Avalible themese for the NIFs
+	var/static/list/ui_themes = list(
+		"abductor",
+		"cardtable",
+		"hackerman",
+		"malfunction",
+		"ntos",
+		"wizard",
+	)
+	///What theme is currently being used on the NIF?
+	var/current_theme = "ntos"
+
+/obj/item/organ/internal/cyberimp/brain/nif/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+
+	if(!ui)
+		ui = new(user, src, "NifPanel", name)
+		ui.open()
+
+/obj/item/organ/internal/cyberimp/brain/nif/ui_state(mob/user)
+	return GLOB.conscious_state
+
+/obj/item/organ/internal/cyberimp/brain/nif/ui_data(mob/user)
+	var/data = list()
+	//User Preference Variables
+	data["linked_mob_name"] = linked_mob.name
+	data["examine_text"] = linked_mob.nif_examine_text
+	data["product_notes"] = manufacturer_notes
+	data["loaded_nifsofts"] = list()
+
+	data["current_theme"] = current_theme
+	data["ui_themes"] = ui_themes
+
+	//NIFSoft Variables
+	data["max_nifsofts"] = max_nifsofts
+
+	for(var/datum/nifsoft/nifsoft as anything in loaded_nifsofts)
```

Comment:
This list appears to be relatively expensive to generate (compared to setting a single value) that you may want to set it in  `ui_static_data()` and then call `update_static_data_for_all_viewers()` on the rare occasion that the loaded_nifsofts is actually changed.

For another example in the code, autolathes and protolathes follow this method and update when their designs change.

This is because `ui_data()` is called every 2 seconds for all open tgui ui's on the server, so it should be for things that are either very cheap to set or constantly change.